src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -445,6 +438,84 @@ extendedKeysOptions()
   return fmt_output.str();
 }
 
+void
+SubCmdValidate::print_help_internal() const
+{
+  if (m_device.empty())
+    printHelp(false, extendedKeysOptions());
+  else {
+    const std::string deviceClass = XBU::get_device_class(m_device, true);
+    auto it = jsonOptions.find(deviceClass);
+  
+    XBUtilities::VectorPairStrings help_tests = { all_test };
+    if (it != jsonOptions.end() && it->second.size() > 3)
+      help_tests.emplace_back(quick_test);
+
+    static const std::string testOptionValues = XBU::create_suboption_list_map(deviceClass, jsonOptions, help_tests);
+    std::vector<std::string> tempVec;
+    common_options.add_options()
+      ("report,r", boost::program_options::value<decltype(tempVec)>(&tempVec)->multitoken(), (std::string("The type of report to be produced. Reports currently available are:\n") + testOptionValues).c_str() )
+    ;
+    printHelp(common_options, m_hiddenOptions, deviceClass, false, extendedKeysOptions());
+  }
+}
+
+static void
+set_test_data(std::vector<std::shared_ptr<TestRunner>>& collection,
+              std::shared_ptr<TestRunner> test,
+              const std::vector<std::string>& param,
+              const std::string& xclbin_path)
+{
+  if (!param.empty() && boost::equals(param[0], test->get_name())) {
+    test->set_param(param[1], param[2]);
+  }
+  if (!xclbin_path.empty())
+    test->set_xclbin_path(xclbin_path);
+
+  collection.push_back(test);
+}
+
+std::vector<std::shared_ptr<TestRunner>>
+SubCmdValidate::collect_and_validate_tests(const std::vector<std::shared_ptr<TestRunner>>& all_tests,
+                                           const std::vector<std::string>& requested_tests,
+                                           const std::vector<std::string>& param,
+                                           const std::string& xclbin_path) const
+{
+  std::vector<std::shared_ptr<TestRunner>> valid_tests;
+
+  if (std::find(requested_tests.begin(), requested_tests.end(), "all") != requested_tests.end()) {
+    for (const auto& test : all_tests) {
+      if (test->getConfigHidden()) 
+        continue;
+
+      set_test_data(valid_tests, test, param, xclbin_path);
+    }
+  } else if (std::find(requested_tests.begin(), requested_tests.end(), "quick") != requested_tests.end()) {
+    if (all_tests.size() < 4)
+      throw xrt_core::error("No test found for: 'quick'\n");
+
+    // Only the first 4 tests are run for a `quick` test
+    for (size_t i = 0; i < 4; i++) {
```

Comment:
This should not have been hard coded, but instead data driven.

Reply:
One day we will figure out something better. For now this is how it has got to be.